cstar/roms/runtime_settings.py

```diff
@@ -0,0 +1,806 @@
+from os import PathLike
+from pathlib import Path
+from typing import ClassVar, Optional, Union, get_args, get_origin
+
+from pydantic import (
+    BaseModel,
+    Field,
+    ModelWrapValidatorHandler,
+    model_serializer,
+    model_validator,
+)
+
+from cstar.base.log import get_logger
+from cstar.base.utils import _list_to_concise_str
+
+log = get_logger(__name__)
+
+################################################################################
+# Formatting methods for serializer:
+
+
+def _format_list_of_floats(float_list: list[float]) -> str:
+    joiner = " "
+    return joiner.join(_format_float(x) for x in float_list)
+
+
+def _format_list_of_paths(
+    path_list: list[Path], multi_line: Optional[bool] = False
+) -> str:
+    joiner = "\n    " if multi_line else " "
+    return joiner.join(_format_path(x) for x in path_list)
+
+
+def _format_list_of_other(other_list: list[str | int]) -> str:
+    joiner = " "
```

Comment:
rename `joiner` to `delimiter` to give var a meaningful name. a "joiner" is something that joins (performs the process) while a delimiter is the thing that goes between two things.